cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -117,7 +116,9 @@ controller, and serviceaccounts controller.`,
 	namedFlagSets := s.Flags(KnownControllers(), ControllersDisabledByDefault.List())
 	verflag.AddFlags(namedFlagSets.FlagSet("global"))
 	globalflag.AddGlobalFlags(namedFlagSets.FlagSet("global"), cmd.Name())
-	cmoptions.AddCustomGlobalFlags(namedFlagSets.FlagSet("generic"))
+	// hoist this flag from the global flagset to preserve the commandline until
+	// the gce cloudprovider is removed.
+	globalflag.Register(namedFlagSets.FlagSet("generic"), "cloud-provider-gce-lb-src-cidrs")
```

Comment:
Should we register this as deprecated (For the KCM)?

Reply:
Done.

Reply:
this wasn't deprecated  in kube-api-server, I assume it should also be deprecated there?

Reply:
It's not deprecated in apiserver yet AFAIK. The cloud controller manager isn't going to fix that.

Reply:
that's unfortunate as it pulls in the cloud provider package to do so (registering without it will panic) :/

do you happen to have any pointers on where to follow-up on that? eventually all the in-tree stuff needs to go.

Reply:
what inside the apiserver depends on the value set by this flag?

Reply:
looks like only the service controller (inside cloud-controller-manager and kube-controller-manager) reference the things that use the variable populated by this flag:

```
lbSrcRngsFlag
    ensureHTTPHealthCheckFirewall
        createTargetPoolAndHealthCheck
            ensureTargetPoolAndHealthCheck
                ensureExternalLoadBalancer
                    EnsureLoadBalancer
                    EnsureLoadBalancerDeleted
                        ServiceController

    LoadBalancerSrcRanges()
        ensureInternalFirewalls
            ensureInternalLoadBalancer
                EnsureLoadBalancer
                EnsureLoadBalancerDeleted
                    ServiceController
```

Reply:
xref: https://github.com/kubernetes/kubernetes/pull/80353
So we should be fine to deprecate and eventually remove the flag (following process of course)?

Reply:
as far as I can see nothing in the apiserver uses the output of this flag. in other cases like this we've just registered a dummy flag connected to a no-op var to live out the deprecation period.

Reply:
thanks, I will raise this in SIG cloud provider